providers/implementations/capabilities/tls.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include "prov/implementations.h"
+#include "prov/provider_ctx.h"
+
+static OSSL_OP_capabilities_supported_fn tls_group_supported;
+
+typedef struct tls_group_info_st {
+    int id;                /* Group ID */
+    const char *name;           /* Group Name */
+    const char *algorithmm;     /* Name of algorithm */
+    int secbits;                /* Bits of security (from SP800-57) */
+} TLS_GROUP_INFO;
+
+static const TLS_GROUP_INFO group_list[] = {
```

Comment:
If we are going to access this array by index in the next table maybe this should set the array size.. [35] =

Reply:
Done